.github/workflows/job-compile-and-test.yml

```diff
@@ -21,10 +21,10 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Use Node.js 16
+      - name: Use Node.js 20
         uses: actions/setup-node@v3

```

Comment:
could we use v4 instead of v3? Any reason to not bump it up?

Reply:
What is v4? How do you know there is a v4?

Reply:
For security reason it is usually a good practice to keep up with latest version of all `@actions/`  actions, like `@actions/setup-node-v4`.  Enabling the dependabot could help in getting automated notifications and PRs to bump those.
You may read the release note to ponder whether that is needed or useful.